TombEngine/Game/effects/effects.cpp

```diff
@@ -787,7 +832,16 @@ void TriggerExplosionSparks(int x, int y, int z, int extraTrig, int dynamic, int
 	if (GetRandomControl() & 1)
 	{
 		if (uw == 1)
-			spark.flags = SP_SCALE | SP_DEF | SP_ROTATE | SP_EXPDEF | SP_UNDERWEXP;
+		{
+			if (mainColor == Vector3::Zero)
```

Comment:
No need to introduce full duplication of spark flags assignment. You can just do:

```if (mainColor == Vector3::Zero)
   spark.flags |= SP_COLOR;
   ```